consul/consul.go

```diff
@@ -203,26 +205,49 @@ func (c *Consul) marathonTaskToConsulService(task *apps.Task, app *apps.App) (*c
 		Port:    task.Ports[0],
 		Address: serviceAddress,
 		Tags:    c.marathonLabelsToConsulTags(app.Labels),
-		Check:   c.marathonToConsulCheck(task, app.HealthChecks, serviceAddress),
+		Checks:  c.marathonToConsulChecks(task, app.HealthChecks, serviceAddress),
 	}, nil
 }
 
-func (c *Consul) marathonToConsulCheck(task *apps.Task, healthChecks []apps.HealthCheck, serviceAddress string) *consulapi.AgentServiceCheck {
-	//	TODO: Handle all types of checks
+func (c *Consul) marathonToConsulChecks(task *apps.Task, healthChecks []apps.HealthCheck, serviceAddress string) consulapi.AgentServiceChecks {
+
+	var checks consulapi.AgentServiceChecks = make(consulapi.AgentServiceChecks, 0, len(healthChecks))
+
 	for _, check := range healthChecks {
-		if check.Protocol == "HTTP" {
-			return &consulapi.AgentServiceCheck{
-				HTTP: (&url.URL{
-					Scheme: "http",
-					Host:   fmt.Sprintf("%s:%d", serviceAddress, task.Ports[check.PortIndex]),
-					Path:   check.Path,
-				}).String(),
+		switch check.Protocol {
+		case "HTTP", "HTTPS":
+
+			if parsedUrl, err := url.Parse(check.Path); err == nil {
+				parsedUrl.Scheme = strings.ToLower(check.Protocol)
+				parsedUrl.Host = fmt.Sprintf("%s:%d", serviceAddress, task.Ports[check.PortIndex])
+
+				checks = append(checks, &consulapi.AgentServiceCheck{
+					HTTP:     parsedUrl.String(),
+					Interval: fmt.Sprintf("%ds", check.IntervalSeconds),
+					Timeout:  fmt.Sprintf("%ds", check.TimeoutSeconds),
+				})
+			}
+		case "TCP":
+			checks = append(checks, &consulapi.AgentServiceCheck{
+				TCP:      fmt.Sprintf("%s:%d", serviceAddress, task.Ports[check.PortIndex]),
 				Interval: fmt.Sprintf("%ds", check.IntervalSeconds),
 				Timeout:  fmt.Sprintf("%ds", check.TimeoutSeconds),
-			}
+			})
+
+		case "COMMAND":
+			checks = append(checks, &consulapi.AgentServiceCheck{
+				Script:   check.Command.Value,
+				Interval: fmt.Sprintf("%ds", check.IntervalSeconds),
+				Timeout:  fmt.Sprintf("%ds", check.TimeoutSeconds),
+			})
+		default:
+			log.WithField("Id", task.AppID.String()).WithField("Address", serviceAddress).
+				Error(fmt.Printf("Unrecognized check protocol %s", check.Protocol))
```

Comment:
We usually use error to indicate request is not processed further. Please log warning here